specutils/io/default_loaders/wcs_fits.py

```diff
@@ -86,3 +91,406 @@ def wcs1d_fits_writer(spectrum, file_name, **kwargs):
+        wcs_dict = {'aperture': aperture,
+                    'beam': beam,
+                    'dtype': disp_type,
+                    'dstart': disp_start,
+                    'avdelt': disp_del_av,
+                    'pnum': pix_num,
+                    'z': dopp_fact,
+                    'alow': aper_low,
+                    'ahigh': aper_high,
+                    'weight': weight,
+                    'zeropoint': zeropoint,
+                    'ftype': function_type,
+                    'order': order,
+                    'pmin': min_pix_val,
+                    'pmax': max_pix_val,
+                    'fpar': params}
+
+        # This section of the code only shows a plot to see if the code
+        # above actually worked which means this methods has not been fully
+        # developed neither tested
+        logging.info('Retrieving model')
+        math_model = _set_math_model(wcs_dict=wcs_dict)
+
+        spectral_axis = math_model(range(1, wcs_dict['pnum'] + 1))
+        return spectral_axis
+
+
+def _set_math_model(wcs_dict):
+    """Defines a mathematical model
+
+    Uses 2 keywords to decided which model is to be built and calls the
+    appropriate function.
+
+    dtype:
+
+        -1 : None, no wavelength solution available
+        0 : Linear wavelength solution
+        1: Log-Linear wavelength solution (not implemented)
+        2: Non-Linear solutions
+            ftype:
+                1: Chebyshev
+                2: Legendre
+                3: Linear Spline (not implemented)
+                4: Cubic Spline (not implemented)
+                5: Pixel Coordinates (not implemented)
+
+    Not implemented models could be implemented on user-request.
+
+
+    Parameters
+    ----------
+
+    wcs_dict : dict
+        Contains all the necessary wcs information needed for building any of
+        models supported.
+
+
+    Returns
+    -------
+
+    `~astropy.modeling.Model`
+
+    """
+    if wcs_dict['dtype'] == -1:
+        return _none()
+    elif wcs_dict['dtype'] == 0:
+        return _linear_solution(wcs_dict=wcs_dict)
+    elif wcs_dict['dtype'] == 1:
+        return _log_linear(wcs_dict=wcs_dict)
+    elif wcs_dict['dtype'] == 2:
+        if wcs_dict['ftype'] == 1:
+            return _chebyshev(wcs_dict=wcs_dict)
+        elif wcs_dict['ftype'] == 2:
+            return _non_linear_legendre(wcs_dict=wcs_dict)
+        elif wcs_dict['ftype'] == 3:
+            return _non_linear_cspline(wcs_dict=wcs_dict)
+        elif wcs_dict['ftype'] == 4:
+            return _non_linear_lspline(wcs_dict=wcs_dict)
+        elif wcs_dict['ftype'] == 5:
+            # pixel coordinates
+            raise NotImplementedError
+        elif wcs_dict['ftype'] == 6:
+            # sampled coordinate array
+            raise NotImplementedError
+        else:
+            raise SyntaxError('ftype {:d} is not defined in the '
+                              'standard'.format(wcs_dict['ftype']))
```

Comment:
Which standard?

Reply:
I could call it iraf standard but I don't know how iraf named it. I only know it does not strictly complies with fits standard defined in the paper I dont remember now.